src/main/java/com/softserve/service/impl/TemporaryScheduleServiceImpl.java

```diff
@@ -195,23 +192,23 @@ public void deleteTemporarySchedulesBySemesterId(Long semesterId) {
     @Override
     public Map<EvenOdd, Map<DayOfWeek, List<TemporarySchedule>>> getTemporaryScheduleForEvenOddWeeks(Long semesterId) {
         log.info("In getTemporaryScheduleForEvenOddWeeks with semesterId = {}", semesterId);
-        Map<EvenOdd, Map<DayOfWeek, List<TemporarySchedule>>> evenOddListMap = new HashMap<>();
+        EnumMap<EvenOdd, Map<DayOfWeek, List<TemporarySchedule>>> evenOddListMap = new EnumMap<>(EvenOdd.class);
         Semester semester = semesterService.getById(semesterId);
         LocalDate today = LocalDate.now();
         LocalDate from = today.with(previousOrSame(MONDAY));
         int countStartDate = semester.getStartDay().getDayOfWeek().getValue();
         int countEndDate = today.getDayOfWeek().getValue();
         int countDays = Integer.parseInt(String.valueOf(ChronoUnit.DAYS.between(
-                semester.getStartDay().minusDays(countStartDate), today.plusDays(7 - countEndDate))));
+                semester.getStartDay().minusDays(countStartDate), today.plusDays(7L - countEndDate))));
 
         EvenOdd keyOne = (countDays / 7) % 2 != 0 ? EvenOdd.EVEN : EvenOdd.ODD;
         EvenOdd keyTwo = keyOne.equals(EvenOdd.EVEN) ? EvenOdd.ODD : EvenOdd.EVEN;
         try {
             List<TemporarySchedule> one = getAllBySemesterAndRange(semesterId, from, from.plusDays(7));
             List<TemporarySchedule> two = getAllBySemesterAndRange(semesterId, from.plusDays(8), from.plusDays(14));
 
-            Map<DayOfWeek, List<TemporarySchedule>> oneMap = new HashMap<>();
-            Map<DayOfWeek, List<TemporarySchedule>> twoMap = new HashMap<>();
+            EnumMap<DayOfWeek, List<TemporarySchedule>> oneMap = new EnumMap<>(DayOfWeek.class);
```

Comment:
No need to change the reference type, it's a kinda rule of thumb to operate with abstraction on the left